app/services/user_associations_service/create.rb

```diff
@@ -36,16 +36,21 @@ def add_user_to_providers
       end
     end
 
-    def add_user_to_provider_in_next_cycle
-      RecruitmentCycle.next.providers.find_by(provider_code: provider.provider_code).users << user
+    def add_user_to_provider_in_both_cycles
+      current_cycle_provider(provider).users << user unless current_cycle_provider(provider).users.include?(user)
+      next_cycle_provider(provider).users << user unless next_cycle_provider(provider).users.include?(user)
```

Comment:
absolutely not a blocker but possible optional refactor in a helper:
```
def add_user_to_provider_in_cycle(cycle_provider)
  cycle_provider.users << user unless cycle_provider.users.include?(user)
end
```

Reply:
Merged this so that we're not blocking users from adding more users. 
```
def add_user_to_provider_in_both_cycles
  add_user_to_provider_in_cycle(current_cycle_provider(provider))
  add_user_to_provider_in_cycle(next_cycle_provider(provider))
end

def add_user_to_provider_in_cycle(cycle_provider)
  cycle_provider.users << user unless cycle_provider.users.include?(user)
end
```

I felt like this was harder to understand. I feel like we needed a `cycle_provider` method which should dynamically determine the recruitment cycle, but we can't do that on the users page as it's not scoped to the recruitment cycle. We probably should do that, as it is on support. Keen to hear your thoughts on this though.